### ISSUE TITLE
[7.9.0] hlsjs downloading unneeded segments when seeking from start (bandwidth issue)

### ISSUE THREAD

**Issue by pixelchutes** [jwplayer/jwplayer#1803]
### Expected Behavior

When beginning playback of HLS stream (via hlsjs) with a `seek` immediately applied, only the necessary segments should be downloaded, allowing near immediate playback.

In `v7.8.6`, this is working as described.

### Actual Behavior

In `v7.9.0`, it does not. A lot of bandwidth is wasted, and a large delay can be seen prior to actual playback.

When beginning playback of HLS stream (via hlsjs) with a `seek` immediately applied, the player immediately fetches the relevant `.ts` segment (e.g. fileSequence123.ts), then proceeds to download _every single segment_ all the way back to **fileSequence0.ts** for the selected bitrate _before_ playback begins.

```php
// Initial seek on start

GET fileSequence196.ts
GET fileSequence195.ts
GET fileSequence194.ts
...
GET fileSequence1.ts
GET fileSequence0.ts

// Playback finally begins
```

In one test, playback was delayed > 2 mins _(and over 200MB!)_

### Steps to reproduce

1. Load HLS stream w/ multiple bitrates (ranging from SD to HD)
2. Execute `seek()` in `on('play')` event, for example, begin 30 mins out.
3. Watch as relevant **.ts** segment gets fetched, but then subsequently downloads all _previous segments_ prior to playback beginning.

#### Example network activity (JW Player 7.9.0)

![hls_out_540](https://cloud.githubusercontent.com/assets/352182/22706255/d2e1dd08-ed2b-11e6-801d-54e0372e2a45.gif)

**NOTE**: Playback does not begin until processing all the way back to very first fileSequence segment.

### Environment

- *macOS 10.12.1*
- *Chrome 56.0.2924.87 (64-bit)*
- *JW Player 7.9.0 (Premium)*

**Comment by johnBartos** [comment]
@pixelchutes This will be fixed in 7.9.1, it's an issue with a backtracking algorithm we implemented which prevents stuttering when keyframes are missing. Can I have a link to your stream?

**Comment by pixelchutes** [comment]
@johnBartos sure thing, can you please provide me an email I can send it to?

**Comment by johnBartos** [comment]
@pixelchutes Sure, email is on my profile

**Comment by pixelchutes** [comment]
Thanks, John. Sent!

**Comment by pixelchutes** [comment]
Confirmed fixed in `7.9.1` on Beta channel. Thank you!

**Comment by johnBartos** [comment]
@pixelchutes Great! Haven't gotten a chance to check your stream but I guess I was right 😁
